lab2/lexic.txt

```diff
@@ -16,19 +16,20 @@
 
 3. Identifiers
 	- a sequence of letters and  digits, such that the first and the last character is a letter; the rule is:
-		identifier = letter | letter{digit}{letter}
+		identifier = letter | letter{(digit|letter)}
 		letter = "A" | "B" | . ..| "Z" | "a" | "b" | ... | "z"
     		digit = "0" | "1" |...| "9"
 
 4. Constants
 	a) integer
 		non_zero_digit = "1" | "2" | ... | "9"
 		digit = "0" | "1" |...| "9"
-		number = {non_zero_digit}non_zero_digit{digit}
+		zero = "0"
+		number = {non_zero_digit}non_zero_digit{digit} | zero
 		signed = ["-"]number
```

Comment:
if you can write positive signed numbers (e.g. +24), you should also add "+" as an option for _signed_. also, the way you wrote it allows for 0 to have a sign. you should write something like _signed_ = ([+|-] number ) | zero (move the _| zero_ part to _signed_.